### PR TITLE
WIP Upgrade kilda-gui on Java 17

### PR DESCRIPTION
**Pull request by oleksir**

### CODE REVIEW

File: src-gui/src/test/java/org/openkilda/controller/FlowControllerTest.java [hunk 1]
```diff
@@ -44,184 +43,184 @@
...
     @Test
-    public void testGetFlowList() {
+        public void testGetFlowList() {
         try {
             mockMvc.perform(get("/api/flows/list").contentType(MediaType.APPLICATION_JSON))
                     .andExpect(status().isOk());
             assertTrue(true);
         } catch (Exception exception) {
-            fail();
+            assertTrue(false);
```
Comment (dmitrii-beliakov) [inline src-gui/src/test/java/org/openkilda/controller/FlowControllerTest.java:69]: What's wrong with the fail() method? Assert true false looks weird.

File: src-gui/src/test/java/org/openkilda/controller/StatsControllerTest.java [hunk 2]
```diff
@@ -15,103 +15,252 @@
...
+    public void testGetFlowStats() throws Exception {
+        try {
+            mockMvc.perform(
+                    get("/api/stats/flowid/{flowId}/{startDate}/{endDate}/{downsample}/{metric}",
+                            TestMockStats.FLOW_ID, TestMockStats.START_DATE,
+                            TestMockStats.END_DATE, TestMockStats.DOWNSAMPLE,
+                            TestMockStats.METRIC_BITS, TestMockStats.DIRECTION_FORWARD)
+                            .contentType(MediaType.APPLICATION_JSON)).andExpect(status().isOk());
+
+            assertTrue(true);
```
Comment (dmitrii-beliakov) [inline src-gui/src/test/java/org/openkilda/controller/StatsControllerTest.java:256]: How is that useful?

**Review by IvanChupin — CHANGES_REQUESTED** [review]
Great job!

1)Could you please consider to use junit jupiter(5) test annotations for test classes instead of annotations from junit 4. (because we no longer use junit 4)

2) Could you please elaborate on the decision to remove `SwitchUtils` class and move its `customSwitchName`() method to switchIntegration service. And what are the benefits to not use `apache.commons.lang3 and collections4` util libraries to check the strings and maps..  because as I see it correctly on Kilda we tend to use apache commons utils libraries wherever it suitable to use.

File: src-gui/src/main/java/org/openkilda/security/SecurityConfig.java [hunk 3]
```diff
@@ -132,53 +79,38 @@ public class SecurityConfig extends WebSecurityConfigurerAdapter {
...
     }
 
-    @Override
-    public void configure(final WebSecurity web) throws Exception {
-        web.ignoring().antMatchers("/resources/**", "/ui/**", "/lib/**");
+    @Bean
+    public WebSecurityCustomizer webSecurityCustomizer() {
+        return web -> web.ignoring()
+                .requestMatchers("/ignore1", "/ignore2");
     }
```
[the remark this answers is not in the document]
Comment (oleksir) [reply src-gui/src/main/java/org/openkilda/security/SecurityConfig.java:107]: it was already there in prev version. So leave it without changes

File: src-gui/src/main/java/org/openkilda/handler/error/MyAccessDeniedHandler.java [hunk 4]
```diff
@@ -38,16 +40,14 @@
 @Component
 public class MyAccessDeniedHandler implements AccessDeniedHandler {
 
-    /**
-     * The logger.
-     */
+    /** The logger. */
     private static Logger logger = LoggerFactory.getLogger(MyAccessDeniedHandler.class);
```
[the remark this answers is not in the document]
Comment (oleksir) [reply src-gui/src/main/java/org/openkilda/handler/error/MyAccessDeniedHandler.java:44]: there are a lot of places where LoggerFactory is in use in kilda-gui. I propose to make it as a separate task.

**Review by IvanChupin — APPROVED** [review]

File: src-gui/src/main/java/org/openkilda/utility/SwitchUtil.java [hunk 5]
```diff
@@ -16,7 +16,7 @@
 package org.openkilda.utility;
 
 import org.apache.commons.collections4.MapUtils;
-import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang.StringUtils;
```
Comment (dmitrii-beliakov) [inline src-gui/src/main/java/org/openkilda/utility/SwitchUtil.java:19]: Why downgrading? lang3 is a more modern version of the commons.

**Review by dmitrii-beliakov — APPROVED** [review]

File: src-gui/src/main/java/org/openkilda/utility/SwitchUtil.java (hunk 5 above)
Comment (oleksir) [reply src-gui/src/main/java/org/openkilda/utility/SwitchUtil.java]: nice catch, replaced in all the places

**Pull request merged**